Data/HashMap/Internal.hs

```diff
@@ -1917,13 +1918,174 @@ differenceCollisions !h1 !ary1 t1 !h2 !ary2
 -- encountered, the combining function is applied to the values of these keys.
 -- If it returns 'Nothing', the element is discarded (proper set difference). If
 -- it returns (@'Just' y@), the element is updated with a new value @y@.
-differenceWith :: (Eq k, Hashable k) => (v -> w -> Maybe v) -> HashMap k v -> HashMap k w -> HashMap k v
-differenceWith f a b = foldlWithKey' go empty a
+differenceWith :: Eq k => (v -> w -> Maybe v) -> HashMap k v -> HashMap k w -> HashMap k v
+differenceWith f = differenceWithKey (const f)
+{-# INLINE differenceWith #-}
+
+-- | \(O(n \log m)\) Difference with a combining function. When two equal keys are
```

Comment:
What are `m` and `n` here?

Reply:
`n` is the size of the first map. `m` is the size of the second map. This is a convention this package uses for many functions. I suspect it was adopted from `containers`.

Reply:
I have opened https://github.com/haskell-unordered-containers/unordered-containers/issues/543 to track this.